device/src/u3v/async_read.rs

```diff
@@ -203,6 +203,9 @@ impl AsyncTransfer {
 impl Drop for AsyncTransfer {
     fn drop(&mut self) {
         unsafe {
+            drop(Box::from_raw(
+                self.transfer().user_data.cast::<AtomicBool>(),
```

Comment:
Just our of curiosity (and my lack of expertise in FFI): this `cast::<AtomicBool>()` does cast it to `*mut AtomicBool`, i.e. a raw pointer to an atomic bool, and not to the `AtomicBool` itself, right? (it likely wouldn't compile anyway, I just find the last of pointer in the generic parameter surprising)

Reply:
That's right :) and yeah, it also surprised me a bit. This is the signature:

```rust
    /// Casts to a pointer of another type.
    #[stable(feature = "ptr_cast", since = "1.38.0")]
    #[rustc_const_stable(feature = "const_ptr_cast", since = "1.38.0")]
    #[inline(always)]
    pub const fn cast<U>(self) -> *mut U {
        self as _
    }
```